unit_test/tstBatchedLinearAlgebra.hpp

```diff
@@ -254,6 +254,466 @@ void matrixTest()
+    EXPECT_EQ( t_view( 0, 0, 0, 1 ), -1.1 );
+    EXPECT_EQ( t_view( 0, 0, 1, 0 ), 4.0 );
+    EXPECT_EQ( t_view( 0, 0, 1, 1 ), 8.7 );
+
+    EXPECT_EQ( t_view( 0, 1, 0, 0 ), 2.0 );
+    EXPECT_EQ( t_view( 0, 1, 0, 1 ), -3.2 );
+    EXPECT_EQ( t_view( 0, 1, 1, 0 ), -6.9 );
+    EXPECT_EQ( t_view( 0, 1, 1, 1 ), -2.1 );
+
+    EXPECT_EQ( t_view( 0, 2, 0, 0 ), -8.3 );
+    EXPECT_EQ( t_view( 0, 2, 0, 1 ), -9.1 );
+    EXPECT_EQ( t_view( 0, 2, 1, 0 ), 3.3 );
+    EXPECT_EQ( t_view( 0, 2, 1, 1 ), -4.4 );
+
+    EXPECT_EQ( t_view( 0, 3, 0, 0 ), 1.4 );
+    EXPECT_EQ( t_view( 0, 3, 0, 1 ), 5.8 );
+    EXPECT_EQ( t_view( 0, 3, 1, 0 ), -5.2 );
+    EXPECT_EQ( t_view( 0, 3, 1, 1 ), -9.1 );
+
+    EXPECT_EQ( t_view( 1, 0, 0, 0 ), 7.2 );
+    EXPECT_EQ( t_view( 1, 0, 0, 1 ), 4.5 );
+    EXPECT_EQ( t_view( 1, 0, 1, 0 ), 4.6 );
+    EXPECT_EQ( t_view( 1, 0, 1, 1 ), 8.8 );
+
+    EXPECT_EQ( t_view( 1, 1, 0, 0 ), -2.5 );
+    EXPECT_EQ( t_view( 1, 1, 0, 1 ), -2.8 );
+    EXPECT_EQ( t_view( 1, 1, 1, 0 ), -1.7 );
+    EXPECT_EQ( t_view( 1, 1, 1, 1 ), 0.3 );
+
+    EXPECT_EQ( t_view( 1, 2, 0, 0 ), 3.1 );
+    EXPECT_EQ( t_view( 1, 2, 0, 1 ), 4.0 );
+    EXPECT_EQ( t_view( 1, 2, 1, 0 ), 0.6 );
+    EXPECT_EQ( t_view( 1, 2, 1, 1 ), -4.8 );
+
+    EXPECT_EQ( t_view( 1, 3, 0, 0 ), -7.7 );
+    EXPECT_EQ( t_view( 1, 3, 0, 1 ), 6.4 );
+    EXPECT_EQ( t_view( 1, 3, 1, 0 ), -9.2 );
+    EXPECT_EQ( t_view( 1, 3, 1, 1 ), 3.1 );
+
+    EXPECT_EQ( t_view( 2, 0, 0, 0 ), -9.0 );
+    EXPECT_EQ( t_view( 2, 0, 0, 1 ), 8.2 );
+    EXPECT_EQ( t_view( 2, 0, 1, 0 ), 7.5 );
+    EXPECT_EQ( t_view( 2, 0, 1, 1 ), 3.4 );
+
+    EXPECT_EQ( t_view( 2, 1, 0, 0 ), 0.3 );
+    EXPECT_EQ( t_view( 2, 1, 0, 1 ), -1.9 );
+    EXPECT_EQ( t_view( 2, 1, 1, 0 ), 9.2 );
+    EXPECT_EQ( t_view( 2, 1, 1, 1 ), -7.7 );
+
+    EXPECT_EQ( t_view( 2, 2, 0, 0 ), -3.5 );
+    EXPECT_EQ( t_view( 2, 2, 0, 1 ), 6.6 );
+    EXPECT_EQ( t_view( 2, 2, 1, 0 ), 3.9 );
+    EXPECT_EQ( t_view( 2, 2, 1, 1 ), 2.9 );
+
+    EXPECT_EQ( t_view( 2, 3, 0, 0 ), 4.0 );
+    EXPECT_EQ( t_view( 2, 3, 0, 1 ), 1.4 );
+    EXPECT_EQ( t_view( 2, 3, 1, 0 ), -6.5 );
+    EXPECT_EQ( t_view( 2, 3, 1, 1 ), -8.2 );
+
+    // Check scalar assignment and () operator
+    t = 43.3;
+    for ( int i = 0; i < 3; ++i )
+    {
+        for ( int j = 0; j < 4; ++j )
+        {
+            for ( int k = 0; k < 2; ++k )
+            {
+                for ( int l = 0; l < 2; ++l )
+                {
+                    EXPECT_EQ( t( i, j, k, l ), 43.3 );
+
+                    t( i, j, k, l ) = -10.2;
+                    EXPECT_EQ( t( i, j, k, l ), -10.2 );
+                }
+            }
+        }
+    }
+
+    // Check default initialization
+    LinearAlgebra::Tensor4<double, 3, 4, 2, 2> s;
+    EXPECT_EQ( s.stride( 0 ), 4 );
+    EXPECT_EQ( s.stride( 1 ), 2 );
+    EXPECT_EQ( s.stride( 2 ), 2 );
+    EXPECT_EQ( s.stride( 3 ), 1 );
+    EXPECT_EQ( s.stride_0(), 4 );
+    EXPECT_EQ( s.stride_1(), 2 );
+    EXPECT_EQ( s.stride_2(), 1 );
```

Comment:
This should be the only change needed to pass tests

Reply:
Done.